lab-davidw/lib/homes.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+module.exports = exports = {};
+
+function Homes(rooms, size) {
+  this.rooms = rooms;
+  this.size = size;
+}
+
+
+exports.Apartment = function(rooms, size) {
+  Homes.call(this, rooms, size);
+  this.isStudio = true;
+};
+
+
+exports.House = function (rooms, size) {
+  Homes.apply(this, [rooms, size]);
+  this.isTownHouse = false;
+};
+
+
+exports.Mansion = function(rooms, size) {
+  Homes.bind(this, rooms, size);
```

Comment:
Generally we use Bind for async functionality. So generally you'd use bind if you wanted to carry context/scope from one function to another.